/*
 *              Copyright (c) 2016-2019, Nuralogix Corp.
 *                      All Rights reserved
 *
 *      THIS SOFTWARE IS LICENSED BY AND IS THE CONFIDENTIAL AND
 *      PROPRIETARY PROPERTY OF NURALOGIX CORP. IT IS
 *      PROTECTED UNDER THE COPYRIGHT LAWS OF THE USA, CANADA
 *      AND OTHER FOREIGN COUNTRIES. THIS SOFTWARE OR ANY
 *      PART THEREOF, SHALL NOT, WITHOUT THE PRIOR WRITTEN CONSENT
 *      OF NURALOGIX CORP, BE USED, COPIED, DISCLOSED,
 *      DECOMPILED, DISASSEMBLED, MODIFIED OR OTHERWISE TRANSFERRED
 *      EXCEPT IN ACCORDANCE WITH THE TERMS AND CONDITIONS OF A
 *      NURALOGIX CORP SOFTWARE LICENSE AGREEMENT.
 */
syntax = "proto3";

option cc_enable_arenas = true;
option java_package = "ai.nuralogix.dfx";
option java_outer_classname = "General";
import "google/protobuf/struct.proto";

package dfx.proto.general;

service Services {
  rpc Status    (StatusRequest)    returns (StatusResponse)    {}
  rpc MimeTypes (MimeTypesRequest) returns (MimeTypesResponse) {}
}

// Status
message StatusRequest {}
message StatusResponse {
  message Note {
    uint32 Date = 1;
    string Message = 2;
  }
  string StatusID = 1;
  string Version = 2;
  map<string, Note> Notes = 3;
}

// Mime Types
message MimeTypesRequest {}
message MimeTypesResponse {
  message MimeType {
    string ID = 1;
    string Name = 2;
    string Mime = 3;
  }
  repeated MimeType Values = 1;
}

// Heartbeat Read
message HeartbeatReadRequest {
  string UUID = 1 ;
}
message HeartbeatReadResponse {
  bool success = 1 ;
  string timestamp = 2 ;
}

// Heartbeat Write
message HeartbeatWriteRequest {
  string UUID = 1 ;
  string Mode = 2 ;
}
message HeartbeatWriteResponse {
  bool success = 1 ;
  google.protobuf.Struct stats = 2 ;
}
